PetaPoco.Tests.Integration/Databases/OracleTests/OracleUpdateTests.cs

```diff
@@ -1,14 +1,114 @@
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using PetaPoco.Core;
+using PetaPoco.Tests.Integration.Models;
 using PetaPoco.Tests.Integration.Providers;
+using Shouldly;
 using Xunit;
 
 namespace PetaPoco.Tests.Integration.Databases.Oracle
 {
-    [Collection("Oracle")]
-    public class OracleUpdateTests : UpdateTests
+    public abstract partial class OracleUpdateTests : UpdateTests
     {
-        public OracleUpdateTests()
-            : base(new OracleTestProvider())
+        protected OracleUpdateTests(TestProvider provider)
+            : base(provider)
         {
         }
+
+        [Collection("Oracle.Delimited")]
+        public class Delimited : OracleUpdateTests
+        {
+            public Delimited()
+                : base(new OracleDelimitedTestProvider())
+            {
+            }
+        }
+
+        [Collection("Oracle.Ordinary")]
+        public class Ordinary : OracleUpdateTests
+        {
+            public Ordinary()
+                : base(new OracleOrdinaryTestProvider())
+            {
+            }
+
+            [Fact]
+            [Trait("Issue", "#667")]
+            public override void Update_GivenTablePrimaryKeyNameAndDynamicType_ShouldNotThrow()
+            {
+                var pd = PocoData.ForType(typeof(Note), DB.DefaultMapper);
+                var tblNote = DB.Provider.EscapeTableName(pd.TableInfo.TableName);
+
+                CreateNote();
+                var entity = DB.Fetch<dynamic>($"SELECT * FROM {tblNote}").First();
+
+#if !NETCOREAPP
+                Should.NotThrow(() => DB.Update("NOTE", "ID", (object)entity));
+#else
+            Should.NotThrow(() => DB.Update("NOTE", "ID", entity));
+#endif
+            }
+
+            [Fact]
+            public override void Update_GivenTablePrimaryKeyNameAndDynamicType_ShouldUpdate()
+            {
+                var pd = PocoData.ForType(typeof(Note), DB.DefaultMapper);
+                var tblNote = DB.Provider.EscapeTableName(pd.TableInfo.TableName);
+
+                CreateNote();
+                var entity = DB.Fetch<dynamic>($"SELECT * FROM {tblNote}").First();
+
+                entity.TEXT += " was updated";
+
+#if !NETCOREAPP
+            DB.Update("NOTE", "ID", (object)entity);
+#else
+                DB.Update("NOTE", "ID", entity);
```

Comment:
I'd suggest doing a quick auto-format with white space diffs enabled in VS btw (may just be this file), for some of the indentation that got shifted.